lib/postgres/postgres-pr/connection.rb

```diff
@@ -213,14 +213,13 @@ def map_compile_arch_to_architecture(compile_arch)
   def detect_platform_and_arch
     result = {}
 
-    query_result = query('select version()').rows.join.match(/on (?<architecture>\w+)-\w+-(?<platform>\w+)/)
-    server_vars = {
-      'version_compile_machine' => query_result[:architecture],
-      'version_compile_os' => query_result[:platform]
-    }
-
-    result[:arch] = map_compile_arch_to_architecture(server_vars['version_compile_machine'])
-    result[:platform] = map_compile_os_to_platform(server_vars['version_compile_os'])
+    query_result = query('select version()').rows[0][0]
+
+    arch = query_result[/on (\w+)-\w+-(\w+)/, 1] || query_result
+    platform = query_result[/on (\w+)-\w+-(\w+)/, 2] || query_result
```

Comment:
I don't think we need to run the same regex on the same input twice, was there an issue with using the named captures like was being done before?

Reply:
+1 to Dean's comment - I think the regex was better before; you just need to handle the nil scenario